mwdb/resources/file.py

```diff
@@ -470,6 +470,124 @@ def post(self, identifier):
         return schema.dump({"token": download_token})
 
 
+@rate_limited_resource
+class FileDownloadResourceXored(Resource):
+    """copy of FileDownloadResource with file_obj.iterate_xored() in .get method"""
```

Comment:
There's awfully lot of code duplication in this class. If I were a maintainer of this project (I'm not) I would be very hesitant to merge this.